# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../hping/hping-bridge.patch
# Copyright (C) 2006 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

This patch fixes hping to run on ethernet bridge interfaces.

It was taken from the gentoo linux repository:
http://sources.gentoo.org/viewcvs.py/gentoo-x86/net-analyzer/hping/files/

--- ./getlhs.c.old	2004-08-05 09:20:35.320759304 -0600
+++ ./getlhs.c	2004-08-05 09:12:40.330968728 -0600
@@ -93,6 +93,9 @@
 	} else if ( strstr(ifname, "eth") ) {
 		linkhdr_size = ETHHDR_SIZE;
 		return 0;
+	} else if ( strstr(ifname, "br") ) {
+		linkhdr_size = ETHHDR_SIZE;
+		return 0;
 	} else if (strstr(ifname, "ets")) {
 		linkhdr_size = 0;
 		return 0;
